src/libraries/System.Data.Common/src/System/Data/DataSet.cs

```diff
@@ -273,6 +278,7 @@ public virtual void GetObjectData(SerializationInfo info, StreamingContext conte
         protected virtual void InitializeDerivedDataSet() { }
 
         // Serialize all the tables.
+        [RequiresUnreferencedCode(RequiresUnreferencedCodeMessage)]
```

Comment:
Correct me if I'm wrong but this is only unsafe if the serializationformat is not xml right since in that case it would use BF. If so, would it make sense to add it to the message so it can be suppressed upstack when we know that the format is xml?

Reply:
No, XML can contain filter expressions which are unsafe